### ISSUE TITLE
✨ Making `StrictlyNegativeInt.Companion.create*` factory functions available on Java

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#439]
## 📝 Description

We would like to make the `StrictlyNegativeInt.Companion.create*` factory functions available for Java users.
This means that their compatibility should be tested and their documentation should indicate their usage from Java code.

## 🔗 Dependencies

This issue is blocked by the following ones:
- [x] #347 
- [x] #434 
- [x] #598 

## ✅ Checklist

- [ ] Make the `create(Number)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `create(Number, () -> Any)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `createOrNull(Number)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Close this issue as completed and update tracking ones if present.

[api binaries]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#checking-the-api-binaries
[test]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#running-tests
[unreleased changelog]: https://github.com/kotools/types/blob/main/CHANGELOG.md#unreleased

**Comment by LVMVRQUXL** [comment]
Another issue for stabilizing these functions will be created while resolving #619.
